screeps-profiler.js

```diff
@@ -146,39 +157,56 @@ const Profiler = {
     if (!Memory.profiler || !Memory.profiler.enabledTick) {
       return 'Profiler not active.';
     }
-
     const elapsedTicks = Game.time - Memory.profiler.enabledTick + 1;
-    const header = 'calls\t\ttime\t\tavg\t\tfunction';
+    const header = 'calls\t\ttime\t\tavg\t\tmax\t\tfunction';
+    const stats = Profiler.stats(Memory.profiler.map);
+    const timeSum = Array.from(stats).reduce((pv, cv) => pv + cv.totalTime, 0);
     const footer = [
       `Avg: ${(Memory.profiler.totalTime / elapsedTicks).toFixed(2)}`,
-      `Total: ${Memory.profiler.totalTime.toFixed(2)}`,
+      `Sum: ${timeSum.toFixed(2)} +${(timeSum - Memory.profiler.lastSum).toFixed(2)}`,
+      `Init: ${Memory.profiler.initTime.toFixed(2)} +${(Memory.profiler.initTime -
+      Memory.profiler.lastInit).toFixed(2)}`,
+      `Total: ${Memory.profiler.totalTime.toFixed(2)} +${(Memory.profiler.totalTime -
+      Memory.profiler.lastTotal).toFixed(2)}`,
       `Ticks: ${elapsedTicks}`,
     ].join('\t');
-    return [].concat(header, Profiler.lines().slice(0, displayresults), footer).join('\n');
+    Memory.profiler.lastTotal = Memory.profiler.totalTime;
+    Memory.profiler.lastSum = timeSum;
+    Memory.profiler.lastInit = Memory.profiler.initTime;
+    return [].concat(header, Profiler.lines(stats).slice(0, displayresults), footer).join('\n');
   },
 
-  lines() {
-    const stats = Object.keys(Memory.profiler.map).map(functionName => {
-      const functionCalls = Memory.profiler.map[functionName];
+  stats(myMap) {
+    const stats = Object.keys(myMap).map(functionName => {
+      const functionCalls = myMap[functionName];
       return {
         name: functionName,
         calls: functionCalls.calls,
-        totalTime: functionCalls.time,
-        averageTime: functionCalls.time / functionCalls.calls,
+        totalTime: functionCalls.time.reduce((pv, cv) => pv + cv, 0),
+        averageTime: functionCalls.time.reduce((pv, cv) => pv + cv, 0) /
+        functionCalls.time.length,
+        maxTime: functionCalls.time.reduce((pv, cv) => Math.max(pv, cv), Number.NEGATIVE_INFINITY),
+        subStats: (functionCalls.parentMap === undefined
+        || Object.keys(functionCalls.parentMap).length < 2)
+        ? null : Profiler.stats(functionCalls.parentMap),
       };
     }).sort((val1, val2) => {
       return val2.totalTime - val1.totalTime;
     });
+    return stats;
+  },
 
+  lines(stats) {
     const lines = stats.map(data => {
       return [
         data.calls,
         data.totalTime.toFixed(1),
         data.averageTime.toFixed(3),
+        data.maxTime.toFixed(3),
```

Comment:
👍 max time sounds awesome, separate PR would be accepted pretty quickly. Make sure to update the docs in the readme as well.